app/src/main/java/org/p2p/wallet/home/ui/new/NewSelectTokenPresenter.kt

```diff
@@ -0,0 +1,118 @@
+package org.p2p.wallet.home.ui.new
+
+import org.p2p.wallet.R
+import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.home.model.SelectTokenItem
+import org.p2p.wallet.home.model.SelectTokenItem.CategoryTitle
+import org.p2p.wallet.home.model.SelectTokenItem.SelectableToken
+import org.p2p.wallet.home.model.SelectableTokenRoundedState.BOTTOM_ROUNDED
+import org.p2p.wallet.home.model.SelectableTokenRoundedState.NOT_ROUNDED
+import org.p2p.wallet.home.model.SelectableTokenRoundedState.ROUNDED
+import org.p2p.wallet.home.model.SelectableTokenRoundedState.TOP_ROUNDED
+import org.p2p.wallet.home.model.Token
+
+private const val QUERY_MIN_LENGTH = 2
+
+class NewSelectTokenPresenter : BasePresenter<NewSelectTokenContract.View>(), NewSelectTokenContract.Presenter {
+
+    private val mappedTokens = mutableListOf<SelectTokenItem>()
+
+    override fun load(tokens: List<Token.Active>, selectedToken: Token.Active?) {
+        mapInitialTokens(selectedToken, tokens)
+
+        view?.showTokens(mappedTokens)
+        view?.showEmptyState(isVisible = mappedTokens.isEmpty())
+    }
+
+    override fun search(lowerCasedSearchText: String) {
```

Comment:
lowerCasedSearchText but maybe add some context that is a query for token?
tokenNameQuery?

Reply:
good one, I like short namings